### PR TITLE
fix(b-modal): transition timing (closes #4761)

### LINKED ISSUE #4761: "show" class is added to modals too late
### Describe the bug

When a modal is shown, it uses a transition, however at the moment `isShow` is set in `onAfterEnter` which doesn't happen until the **end** of the transition. This means you get sequential transitions or a perceived delay, also the buttons which should show disabled during the transition are always enabled.

### Steps to reproduce the bug

1. Extend the transition time of `fade` to make it easier to see
2. Trigger open of a modal
3. You will see the backdrop fade in
4. Then after that the `show` class is added and you see the modal show transition

### Expected behavior

The `show` class needs to be added at the same point as a `enter-to` class would be added, i.e. the very next frame after enter.

### Versions

**Libraries:**

- BootstrapVue: 2.4.1
- Bootstrap: 4.4.1
- Vue: 2.6.10

### Fix

To solve the issue I changed the following here: https://github.com/bootstrap-vue/bootstrap-vue/blob/7a34f737be4825bfc1313463b13e2edb529f9a7c/src/components/modal/modal.js#L616

    onEnter() {
        this.isBlock = true;
        window.requestAnimationFrame(() => {
            window.requestAnimationFrame(() => {
                this.isShow = true;
            });
        });
    }

And also remove `this.isShow = true;` from `onAfterEnter`

Ideally I'd like to use `nextFrame(() => {this.isShow = true;})` from the built in transition lib in Vue but I couldn't see a way to import that definition. https://github.com/vuejs/vue/blob/4de4649d9637262a9b007720b59f80ac72a5620c/src/platforms/web/runtime/transition-util.js#L67

### Demo link for problem and solution

https://codepen.io/sonicisthebest/pen/LYVGpvJ

### PR DESCRIPTION
**Pull request by tmorehouse**
### Describe the PR

Minor adjustments to the modal show enter transition timing

Closes #4761

### PR checklist

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Enhancement
- [ ] ARIA accessibility
- [ ] Documentation update
- [ ] Other (please describe)

**Does this PR introduce a breaking change?** (check one)

- [x] No
- [ ] Yes (please describe)

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch, **not** the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (i.e. `[...] (fixes #xxx[,#xxx])`, where "xxx" is the issue number)
- [x] It should address only one issue or feature. If adding multiple features or fixing a bug and adding a new feature, break them into separate PRs if at all possible.
- [x] The title should follow the [**Conventional Commits**](https://www.conventionalcommits.org/) naming convention (i.e. `fix(alert): not alerting during SSR render`, `docs(badge): update pill examples`, `chore(docs): fix typo in README`, etc). **This is very important, as the `CHANGELOG` is generated from these messages.**

**If new features/enhancement/fixes are added or changed:**

- [ ] Includes documentation updates (including updating the component's `package.json` for slot and event changes)
- [ ] Includes any needed TypeScript declaration file updates
- [ ] New/updated tests are included and passing (if required)
- [x] Existing test suites are passing
- [x] CodeCov for patch has met target
- [x] The changes have not impacted the functionality of other components or directives
- [x] ARIA Accessibility has been taken into consideration (Does it affect screen reader users or keyboard only users? Clickable items should be in the tab index, etc.)

**If adding a new feature, or changing the functionality of an existing feature, the PR's
description above includes:**

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

### CODE REVIEW

File: src/components/modal/modal.js [hunk 1]
```diff
@@ -604,10 +604,13 @@ export const BModal = /*#__PURE__*/ Vue.extend({
     },
     onEnter() {
       this.isBlock = true
+      // We add show class 1 frame after
+      requestAF(() => {
```
Comment (jackmu95) [inline src/components/modal/modal.js]: @tmorehouse Is one `requestAF()` enough here?

Comment (tmorehouse) [reply src/components/modal/modal.js]: Yup.

Just enough to trigger the browser CSS show animation/transitions to start - one frame after the content is inserted (or display:none removed)

Comment (tmorehouse) [reply src/components/modal/modal.js]: Vue transition `enter-class` is added one frame after insert (or removal of display: none)

Comment (tmorehouse) [reply src/components/modal/modal.js]: But we can't use enter-to-class, as Vue immediately removes it after the enter transition completes.

Comment (jackmu95) [reply src/components/modal/modal.js]: Ok 👍 I've tried @philipmountifield's demo and he had to in there to mimic the `nextFrame()` util.

**Pull request merged**